app/uk/gov/hmrc/exports/scheduler/jobs/SendEmailsJob.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.scheduler.jobs
+
+import java.time.LocalTime
+
+import javax.inject.{Inject, Named, Singleton}
+import org.joda.time.DateTime
+import play.api.Logging
+import uk.gov.hmrc.exports.config.AppConfig
+import uk.gov.hmrc.exports.models.emails.SendEmailDetails
+import uk.gov.hmrc.exports.models.emails.SendEmailResult._
+import uk.gov.hmrc.exports.repositories.SendEmailWorkItemRepository
+import uk.gov.hmrc.exports.services.email.EmailSender
+import uk.gov.hmrc.workitem.{Failed, Succeeded, WorkItem}
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class SendEmailsJob @Inject()(appConfig: AppConfig, sendEmailWorkItemRepository: SendEmailWorkItemRepository, emailSender: EmailSender)(
+  implicit @Named("jobsExecutionContext") ec: ExecutionContext
+) extends ScheduledJob with Logging {
+
+  override def name: String = "SendEmails"
+
+  override def firstRunTime: LocalTime = {
+    val now = LocalTime.now(appConfig.clock).withSecond(0).withNano(0)
+    calcRunTime(now)
+  }
+
+  @scala.annotation.tailrec
+  private def calcRunTime(time: LocalTime): LocalTime = {
+    val proposedTime = time.plusMinutes(1)
+    if (proposedTime.getMinute % interval.toMinutes == 0)
+      proposedTime
+    else
+      calcRunTime(proposedTime)
+  }
+
+  override def interval: FiniteDuration = appConfig.sendEmailsJobInterval
+
+  override def execute(): Future[Unit] = {
+    logger.info("Starting SendEmailsJob execution...")
+
+    val now = DateTime.now()
+    val failedBefore = interval.minus(1.minute).toMinutes
+
+    def process(): Future[Unit] =
+      sendEmailWorkItemRepository.pullOutstanding(failedBefore = now.minusMinutes(failedBefore.toInt), availableBefore = now).flatMap {
```

Comment:
According to the docs failedBefore means 'it will only consider WorkItems in FailedState if they were marked as Failed before the failedBefore dateTime'. Should this not be configurable (like the scheduled job run interval)? 

Just thinking about the scenario where if (for whatever reason) a particular email requests causes a 500 response from the email server (perhaps they have a bug and can not handle a particularly odd email address) but other emails could be sent. If on the first run attempt this email fails to be sent (and stops the scheduledJob processing other emails) it would be good to exclude this email from the next scheduledJob run to allow the other emails a chance to be sent. 

So setting this failedBefore to be greater than the scheduled job run interval would achieve this (by resulting in any failed email not being reattempted in the very next process run at least).

Reply:
For the scenario you described the Job would need to go through other emails as well in case of 500 response, actually. Otherwise you risk having 2 emails blocking the all others if they are picked up every second run.
Not sure if this may happen though. In the ticket it is specified that for 5xx response the Job should stop until next run time.

I will make it configurable, but I'll leave it's value and 5xx response handling as it is now. This requires wider discussion IMHO.